airbyte-integrations/connectors/destination-kafka/src/main/resources/spec.json

```diff
@@ -0,0 +1,223 @@
+      "transactional_id": {
+        "title": "Transactional ID",
+        "description": "The transactional ID to use for transactional delivery. This enables reliability semantics which span multiple producer sessions since it allows the client to guarantee that transactions using the same transactional ID have been completed prior to starting any new transactions. If no transactional ID is provided, then the producer is limited to idempotent delivery.",
+        "type": "string",
+        "default": "",
+        "examples": ["txn-id"],
+        "order": 9
+      },
+      "transaction_timeout_ms": {
+        "title": "Transaction timeout ms",
+        "description": "The maximum amount of time in ms that the transaction coordinator will wait for a transaction status update from the producer before proactively aborting the ongoing transaction.",
+        "type": "integer",
+        "default": 60000,
+        "order": 10
+      },
+      "enable_idempotence": {
+        "title": "Enable idempotence",
+        "description": "When set to 'true', the producer will ensure that exactly one copy of each message is written in the stream. If 'false', producer retries due to broker failures, etc., may write duplicates of the retried message in the stream.",
+        "type": "boolean",
+        "default": false,
+        "order": 11
+      },
+      "compression_type": {
+        "title": "Compression type",
+        "description": "The compression type for all data generated by the producer.",
+        "type": "string",
+        "default": "none",
+        "enum": ["none", "gzip", "snappy", "lz4", "zstd"],
+        "order": 12
+      },
+      "batch_size": {
+        "title": "Batch size",
+        "description": "The producer will attempt to batch records together into fewer requests whenever multiple records are being sent to the same partition.",
+        "type": "integer",
+        "default": 16384,
+        "order": 13
+      },
+      "linger_ms": {
+        "title": "Linger ms",
+        "description": "The producer groups together any records that arrive in between request transmissions into a single batched request.",
+        "type": "integer",
+        "default": 0,
+        "order": 14
+      },
+      "max_in_flight_requests_per_connection": {
+        "title": "Max in flight requests per connection",
+        "description": "The maximum number of unacknowledged requests the client will send on a single connection before blocking.",
+        "type": "integer",
+        "default": 5,
+        "order": 15
+      },
+      "client_dns_lookup": {
+        "title": "Client DNS lookup",
+        "description": "Controls how the client uses DNS lookups.",
+        "type": "string",
+        "default": "use_all_dns_ips",
+        "enum": [
+          "default",
+          "use_all_dns_ips",
+          "resolve_canonical_bootstrap_servers_only",
+          "use_all_dns_ips"
+        ],
+        "order": 16
+      },
+      "buffer_memory": {
```

Comment:
can we relieve the user of needing to think about this by choosing a sane default?

Reply:
We already have a default value